### PR TITLE
Lib navball improvements

### PR DESCRIPTION
**Pull request by nuggreat**
This should resolve the point I raised in issue #121

### CODE REVIEW

**Comment by scimas** [comment]
Could you also make `ves` default to `ship`? That sounds like a very sane default for the parameter, likely what most users would use it for anyway.

File: library/lib_navball.ks [hunk 1]
```diff
@@ -11,44 +11,111 @@ function east_for {
...
+  parameter ves,thing is "default".
+
+  local pointing is ves:facing.
+  if not thing:istype("string") {
+    if thing:istype("vessel") or pointing:istype("part") {
+      set pointing to thing:facing.
+    } else if thing:istype("direction") {
+      set pointing to thing.
+    } else {
+      print "type: " + thing:typename + " is not reconized by lib_navball".
```
Comment (scimas) [inline library/lib_navball.ks]:
```suggestion
      print "type: " + thing:typename + " is not reconized by roll_for".
```
This is specific for this function only.

File: library/lib_navball.ks [hunk 2]
```diff
@@ -11,44 +11,111 @@ function east_for {
...
-    }
-  } 
-}.
+    local vec_y is vcrs(ves:up:vector,ves:facing:forevector).
+    local trig_y is vdot(pointing:topvector,vec_y).
+    return arctan2(trig_y,trig_x).
+  }
+}
+
+function compass_and_pitch_for {
```
Comment (scimas) [inline library/lib_navball.ks]: Any reason for not simply creating a wrapper around `compass_for` and `pitch_for` here?

File: doc/lib_navball.md [hunk 3]
```diff
@@ -21,34 +21,88 @@ description:
...
+    * if not a vector arg_2 will be converted into a vector by the function ``type_to_vector``
+  3) can be a vector,direction,vessel,part,or geoposition.
+    * if not a vector arg 3 will be converted into a vector by the function ``type_to_vector``
+
+returns:
+  *  a number of degrees in the range [-180..180]
+
+description:
+  * Returns the relative heading of arg 3, with arg_2 treated as north, with the horizon defined by arg_1.
+    Will be positive if arg 3 to the "east" of arg_2, and negative if arg 3 is to the "west" of arg_2
```
Comment (scimas) [inline doc/lib_navball.md]:
```suggestion
    Will be positive if arg_3 is to the "east" of arg_2, and negative if arg_3 is to the "west" of arg_2
```

Comment (nuggreat) [reply doc/lib_navball.md]: done

File: library/lib_navball.ks (hunk 2 above)
Comment (nuggreat) [reply library/lib_navball.ks]: The reason why I didn't just wrapper `compass_for` and `pitch_for` is that the standalone function is about 20% faster than if I had wrapped the other 2.

File: library/lib_navball.ks (hunk 1 above)
Comment (nuggreat) [reply library/lib_navball.ks]: done

**Comment by nuggreat** [comment]
I also did default `ves` to `ship` where possible and updated the documentation to reflect that.

File: doc/lib_navball.md [hunk 4]
```diff
@@ -21,34 +21,88 @@ description:
...
-  * Returns the pitch above the horizon for the vessel's nose.
+  * Returns the pitch above the horizon as measured from the vessel defined by arg_1 of arg_2.
     Pitching up gets a positive number, Pitching down a negative number.
 
 ### roll_for
 
 args:
-  * a Vessel, such as ``SHIP``.
+  1) a Vessel, such as ``SHIP``.
+  2) an optional argument defaulted to ``arg1:facing``, can be a direction,vessel, or part.
```
Comment (scimas) [inline doc/lib_navball.md]:
```suggestion
  2) an optional argument defaulted to ``arg1:facing``, can be a direction, vessel, or part.
```

**Comment by scimas** [comment]
Ready for merge after that small formatting change.

File: doc/lib_navball.md (hunk 4 above)
Comment (nuggreat) [reply doc/lib_navball.md]: done

**Pull request merged**